src/client/address_book/impl.h

```diff
@@ -63,6 +63,99 @@ class AddressBookSubscriber;
 /// @brief Address book implementation
 class AddressBook : public AddressBookDefaults {
  public:
+  /// @brief Container for Address Book entry
+  class Entry
+  {
+   public:
+    /// @brief No default copy-constructor
+    Entry(Entry& rhs) = delete;
+    /// @brief No default move-constructor
+    Entry(Entry&& rhs) = delete;
+    /// @brief Create address book entry from a hostname & identity hash
+    /// @param host Kovri hostname
+    /// @param address Kovri identity hash
+    explicit Entry(
+        const std::string&& host,
+        const kovri::core::IdentHash&& address)
+    {
+      try
+        {
+          if (host.empty())
+            throw std::runtime_error("AddressBook: empty entry hostname");
+          m_Host = host;
+          m_Address = address;
+        }
+      catch (...)
+        {
+          kovri::core::Exception ex;
+          ex.Dispatch(__func__);
+          throw;
+        }
+    }
+
+    /// @brief Create address book entry from a host & base64-encoded address
+    /// @param host Kovri hostname
+    /// @param address Base64-encoded Kovri address
+    explicit Entry(const std::string&& host, const std::string& address)
+        : m_Host(host)
+    {
+      try
+        {
+          if (m_Host.empty())
+            throw std::runtime_error("AddressBook: empty entry hostname");
+          kovri::core::IdentityEx ident;
+          ident.FromBase64(address);
+          m_Address = std::move(ident.GetIdentHash());
+        }
+      catch (...)
+        {
+          kovri::core::Exception ex;
+          ex.Dispatch(__func__);
+          throw;
+        }
+    }
+
+    /// @brief Create address book entry from a subscription line
+    /// @param subscription_line Line from a subscription file (host=address format)
+    /// @notes Throws on malformed subscription lines
+    explicit Entry(const std::string& subscription_line)
+    {
+      try
+        {
+          std::size_t pos = subscription_line.find('=');
+          if (pos == std::string::npos)
+            throw std::runtime_error("AddressBook: invalid subscription line");
+          m_Host = std::move(subscription_line.substr(0, pos++));
+          if (m_Host.empty())
+            throw std::runtime_error("AddressBook: empty entry hostname");
+          kovri::core::IdentityEx ident;
+          ident.FromBase64(subscription_line.substr(pos));
+          m_Address = std::move(ident.GetIdentHash());
```

Comment:
```cpp
m_Address = ident.GetIdentHash();
```
IdentytyEx is trivially-copyable https://clang.llvm.org/extra/clang-tidy/checks/performance-move-const-arg.html